lib/src/git.dart

```diff
@@ -153,7 +163,7 @@ bool _tryGitCommand(String command) {
       // We just warn here, as some features might work with older versions of
       // git.
       log.warning('''
-You have a very old version of git (version ${output.substring('git version '.length)}),
+You have a very old version of git (version ${versionString.substring('git version '.length)}),
 for $topLevelProgram it is recommended to use git version 2.14 or newer.
```

Comment:
Could you change this to:
```suggestion
You have a very old version of git (version ${major}.${minor}),
for $topLevelProgram it is recommended to use git version 2.14 or newer.
```
And then I don't think we need the `versionString` variable.

Reply:
Glad to do this.

Reply:
Ah - from the failed test I see why we had to do it the other way... The version string can be longer than just major.minor.

So I take my suggestion back. Could you revert, but please set the `versionString` to the substring value when it is stored (so the name fits better). Also it seems it should not be needed to check both `match` and `versionString` against null, just one of them.

Reply:
Done